packages/pg-protocol/src/parser.ts

```diff
@@ -284,7 +286,12 @@ export class Parser {
     for (let i = 0; i < fieldCount; i++) {
       const len = this.reader.int32()
       // a -1 for length means the value of the field is null
-      fields[i] = len === -1 ? null : this.reader.string(len)
+      if (len === -1) {
+        fields[i] = null
+      } else {
+        const format = this.fieldDescriptions[i]?.format
```

Comment:
i don't see any other code in pg-protocol doing this, but i would prefer to `assert(this.fieldDescriptions[i], 'Missing field description');` instead of defaulting to text format if a field description is not found.